USER_GUIDE.md

```diff
@@ -359,7 +360,64 @@ try {
 
 ```
 
-## ClientBuilder
+## Client Factories
```

Comment:
User guide has a TOC, update that too pls.

Reply:
Updated